From a09fd5f7afbb7dc9ac336a60928b8e2aa93da41c Mon Sep 17 00:00:00 2001
From: Mitul Chokshi <mitul.chokshi@intel.com>
Date: Tue, 2 May 2017 15:40:21 +0000
Subject: [PATCH 1367/2313] [INTERNAL] drm/i915: properly add vgt-id in context
 id

It seems that some part of the original patch of commit 576ab26a2bba
("drm/i915/gvt: Add vgt-id in context id") was missing during rebase.
This patch fixed the missing part.

Signed-off-by: Mitul Chokshi <mitul.chokshi@intel.com>
Signed-off-by: Min He <min.he@intel.com>
Reviewed-by: Singh, Satyeshwar <satyeshwar.singh@intel.com>
---
 drivers/gpu/drm/i915/i915_gem_context.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/drivers/gpu/drm/i915/i915_gem_context.c b/drivers/gpu/drm/i915/i915_gem_context.c
index 1895687..e4cb164 100644
--- a/drivers/gpu/drm/i915/i915_gem_context.c
+++ b/drivers/gpu/drm/i915/i915_gem_context.c
@@ -90,6 +90,7 @@
 #include <drm/i915_drm.h>
 #include "i915_drv.h"
 #include "i915_trace.h"
+#include "i915_vgpu.h"
 
 #define ALL_L3_SLICES(dev) (1 << NUM_L3_SLICES(dev)) - 1
 
@@ -230,6 +231,12 @@ static int assign_hw_id(struct drm_i915_private *dev_priv, unsigned *out)
 			return ret;
 	}
 
+	if (intel_vgpu_active(dev_priv)) {
+		/* add vgpu_id to context hw_id */
+		ret = ret | (I915_READ(vgtif_reg(vgt_id))
+				<< SIZE_CONTEXT_HW_ID);
+	}
+
 	*out = ret;
 	return 0;
 }
-- 
2.7.4

